Lib/fontParts/objects/base/validators.py

```diff
@@ -115,10 +115,11 @@ def validateLayerName(value):
 # -----
 
 def validateGlyphName(value):
-    """
-    XXX implement
-    """
-    return value
+    if not isinstance(value, basestring):
+        raise FontPartsError("Glyph names must be unicode strings, not %s." % type(value).__name__)
```

Comment:
It should accept either `str` or `uni` (thus `basestring`). This way scripters in 2.7 won't have to update their scripts to do this `glyph.name = u"foo"`. That would cause a lot of breakage. The validator should convert a `str` to `uni` and return that. The validators also normalize types so that environment implementations don't have to do that work.